src/writer/flag-proposal.ts

```diff
@@ -0,0 +1,35 @@
+import { jsonParse } from '../helpers/utils';
+import db from '../helpers/mysql';
+import { getSpace, getProposal } from '../helpers/actions';
+
+export function isAuthorized({ space, address }): boolean {
+  const admins = (space?.admins || []).map(admin => admin.toLowerCase());
+  const mods = (space?.moderators || []).map(mod => mod.toLowerCase());
+
+  return admins.includes(address.toLowerCase()) || mods.includes(address.toLowerCase());
+}
+
+export async function verify(body): Promise<any> {
+  const msg = jsonParse(body.msg);
+
+  const proposal = await getProposal(msg.space, msg.payload.proposal);
+  const space = await getSpace(msg.space);
+  if (!proposal) return Promise.reject('unknown proposal');
+
+  const isAuthorizedToFlag = isAuthorized({
+    space,
+    address: body.address
+  });
+  if (!isAuthorizedToFlag) return Promise.reject('not authorized to flag proposal');
```

Comment:
Could we also check if the proposal is already flagged (as we already got the proposal info) better to reject it ❎ but not too important

Reply:
done on [line](https://github.com/snapshot-labs/snapshot-sequencer/pull/168/files#diff-3d4aaa652e292913a8c281d6a8d624a427f621dba27479cc2ad0940ee089c684R17)